using_images/s2i_images/java.adoc

```diff
@@ -10,93 +10,191 @@
 
 toc::[]
 
+[[s2i-images-java-overview]]
 == Overview
-{product-title} provides
-xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[S2I]
-enabled Java images for building and running Java applications.
-ifdef::openshift-origin[]
-The https://github.com/fabric8io/java-main/tree/master/[Java S2I builder image]
-endif::openshift-origin[]
-ifdef::openshift-enterprise[]
-The Java S2I builder image
-endif::openshift-enterprise[]
-assembles your application source with any required dependencies to create a new
-image containing your Java application. This resulting image can be run either
-by {product-title} or by Docker. This image is intended for use with
-https://maven.apache.org[Maven]-based Java standalone projects (that are run via
-main class).
 
+{product-title} provides an
+xref:../../architecture/core_concepts/builds_and_image_streams.adoc#source-build[S2I
+builder image] for building Java applications.  This builder image takes your
+application source or binary artifacts, builds the source using Maven (if source
+was provided), and assembles the artifacts with any required dependencies to
+create a new, ready-to-run image containing your Java application. This
+resulting image can be run on {product-title} or run directly with Docker.
+
+The builder image is intended for use with
+link:https://maven.apache.org[Maven]-based Java standalone projects that are run
+via main class.
+
+
+[[s2i-images-java-versions]]
 == Versions
-The current version supports JDK 1.7 and Maven 3.2.x projects.
 
+The current version of the Java S2I builder image supports OpenJDK 1.8, Jolokia
+1.3.5, and Maven 3.3.9-2.8.
+
+
+[[s2i-images-java-images]]
 == Images
 
-To use this image, you can either access it directly from those
-xref:../../architecture/infrastructure_components/image_registry.adoc#architecture-infrastructure-components-image-registry[image
-registries], or push it into your
-xref:../../install_config/registry/index.adoc#install-config-registry-overview[{product-title} Docker registry].
-Additionally, you can create an
+The RHEL 7 image is available through Red Hat's subscription registry:
```

Comment:
s/Red Hat's subscription registry/the Red Hat Registry/

See https://access.redhat.com/containers/#/faq.